### DISCUSSION TITLE
GitVersion does not handle repos without branches named develop or main/master

### DISCUSSION THREAD

**Discussion by CasperWSchmidt** [GitTools/GitVersion#3669] [Q&A] [answered]
As stated in another [discussion](https://github.com/GitTools/GitVersion/discussions/3663#discussioncomment-6801652) we've decided on an unusual branching strategy where neither main/master nor develop is present. We've coined it multiple-trunk-based development.

During testing of this setup I've discovered that branching from _release/v1_ branch (that is set as `is-mainline: true`, `is-release-branch: true` and `mode: ContinuousDelivery`) to _feature/someFeature_ is unable to produce a version for the feature branch. It simply fails with an error desciption telling me it wasn't able to find either develop or main/master. I would expect GetVersion to use the `is-mainline` to determine which branch is the main/master branch and use that. Instead it looks like the names develop and main/master is hardcoded into the tool.

Am I right about this assumption? Is there any way to get around this to support our strategy?

NB: For now we continue with the old strategy using develop and main (V1 is still in alpha so there's obviously no V2 yet)

**Comment by HHobeck** [comment, marked as answer]
Hi there.

If you take the following integration test then you see it is working out of the box:

```csharp
[Test]
public void TestForDiscussion3669()
{
    var configuration = GitFlowConfigurationBuilder.New
        .WithBranch("release", _ => _
            .WithIncrement(IncrementStrategy.Minor)
            .WithIsMainline(true)
            .WithIsReleaseBranch(true)
            .WithVersioningMode(VersioningMode.ContinuousDelivery)
        ).Build();

    using var fixture = new EmptyRepositoryFixture("release/v1");

    fixture.MakeATaggedCommit("1.0.0");
    fixture.BranchTo("feature/just-a-feature");
    fixture.MakeACommit();

    // ✅ succeeds as expected
    fixture.AssertFullSemver("1.1.0-just-a-feature.1+1", configuration);
}
```

The is-mainline branch property is only used to detect commit tags on branches which are not present in the current commit line.
